index.bs

```diff
@@ -2604,23 +2774,36 @@ To <dfn>process an attribution source</dfn> given an [=attribution source=] |sou
     to the user agent's [=max pending sources per source origin=]:
     1. Run [=obtain and deliver a debug report on source registration=] with "<code>[=source debug data type/source-storage-limit=]</code>" and |source|.
     1. Return.
-1. If the result of running [=check if an attribution source exceeds the unexpired destination limit=]
-    with |source| is true:
-    1. Run [=obtain and deliver a debug report on source registration=] with "[=source debug data type/source-destination-limit=]</code>" and |source|.
-    1. Return.
-1. If the result of running [=check if an attribution source should be blocked by reporting-origin per site limit=]
-    with |source| is <strong>blocked</strong>:
-    1. Run [=obtain and deliver a debug report on source registration=] with "<code>[=source debug data type/source-reporting-origin-per-site-limit=]</code>" and |source|.
-    1. Return.
 1. Let |destinationRateLimitResult| be the result of running [=check if an attribution source exceeds the time-based destination limit=] with |source|.
 1. If |destinationRateLimitResult| is "<code>[=destination rate-limit result/hit reporting limit=]</code>":
     1. Run [=obtain and deliver a debug report on source registration=] with "<code>[=source debug data type/source-destination-rate-limit=]</code>" and |source|.
     1. Return.
+1. If the result of running [=check if an attribution source exceeds the per day destination limit=]
+    with |source| is true:
+    1. Run [=obtain and deliver a debug report on source registration=] with "<code>[=source debug data type/source-destination-per-day-rate-limit=]</code>" and |source|.
+    1. Return.
+1. Let |sourcesToDeleteForDestinationLimit| be the result of running [=get sources to delete for the unexpired destination limit=]
+    with |source|.
+1. Let |destinationLimitHit| be false if |sourcesToDeleteForDestinationLimit| [=set/is empty=], true otherwise.
+1. Let |dropNewSource| be false.
+1. If |source|'s [=attribution source/destination limit algorithm=] is:
+    <dl class="switch">
+    : "<code>[=destination limit algorithm/priority_fifo=]</code>"
+    :: Set |dropNewSource| to true if |sourcesToDeleteForDestinationLimit| [=set/contains=] |source|'s [=attribution source/source identifier=].
+    : "<code>[=destination limit algorithm/lifo=]</code>"
+    :: Set |dropNewSource| to |destinationLimitHit|.
+
+    </dl>
+1. If |dropNewSource| is true:
+    1. Run [=obtain and deliver a debug report on source registration=] with "[=source debug data type/source-destination-limit=]</code>" and |source|.
+    1. Return.
+1. Run [=delete sources for unexpired destination limit=] with |sourcesToDeleteForDestinationLimit| and |source|'s [=attribution source/source time=].
```

Comment:
Do we need to a conditional deletion here based on the `dropNewSource` flag?

It seems that the newest source won't be deleted in case of LIFO, or am I missing something?

Let's say the limit is 1 and there are 2 sources: oldSource{destination: d1} & newSource {destination: d2}.

- Step 4 in `get sources to delete for the unexpired destination limit` the sorted list will look like [d2, d1].
- Step 6 in  get sources to delete for the unexpired destination limit` will add oldSource to `sourcesToDelete` instead of newSource (in case of LIFO)

Reply:
If `|dropNewSource|` is true, it will early return. Therefore this is only reachable when `|dropNewSource|` is false, that is, we only delete sources if the new source is not to be dropped by this limit.

Reply:
And in case of FIFO, dropNewSource will be true as long as the returning set is non empty, so new source will be deleted.

Reply:
Ack. I missed the early return.